src/App.jsx

```diff
@@ -14,45 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'alp';
+const SORT_BY_LENGTH = 'lng';
+const REVERSE = 'rev';
+
+function sortGoods(goods, { sortCondition, reverseGoods }) {
+  const sortedGoods = [...goods];
+
+  if (sortCondition) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortCondition) {
+        case SORT_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseGoods) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [sortCondition, setSortCondition] = useState('');
+  const [reverseGoods, setReverseGoods] = useState('');
```

Comment:
The initial state of 'reverseGoods' is a string, which is not intuitive. Consider using Boolean for this state. Also, you should initialize this to false since there is no need to reverse the goods at the beginning.